File: leetcode/triangle/triangle.py
class Solution:
    def minimumTotal(self, triangle: List[List[int]]) -> int:
        for lvl in range(len(triangle)):
            for marker in range(len(triangle[lvl-1])):
                triangle[lvl-1][marker] += min(triangle[lvl][marker:marker+2])

        return triangle[0][0]

File: leetcode/triangle/triangle.cpp
class Solution {
public:
	int minimumTotal(vector<vector<int>>& triangle) {

		for (int lvl = triangle.size() - 1; lvl > 0; lvl--) {
			for (int marker = 0; marker < triangle[lvl-1].size(); marker++) {
				
				int min_in_pair = triangle.at(lvl).at(marker);
				if (triangle.at(lvl).at(marker+1) < min_in_pair)
					min_in_pair = triangle.at(lvl).at(marker+1);

				triangle.at(lvl-1).at(marker) += min_in_pair;
			}
		}

		return triangle.at(0).at(0);
	}
};
